include/deal.II/lac/sparsity_tools.h

```diff
@@ -103,6 +103,33 @@ namespace SparsityTools
                   const Partitioner partitioner = Partitioner::metis
                  );
 
+  /**
+   * Using a coloring algorithm provided by ZOLTAN to color indices whose
+   * connections are represented using a SparsityPattern object. In effect,
+   * we view this sparsity pattern as a graph of connections between nodes,
+   * where each nonzero entry in the sparsity pattern corresponds to
+   * an edge between two nodes in the connection graph. The goal is to assign
+   * each node a color index such that no two connected nodes have
+   * same color index. Note that this function requires that SparsityPattern be
+   * symmetric (and hence square) i.e a bidirected graph representation. We do
+   * not check for symmetry of the sparsity pattern, since this is an
+   * expensive operation, but rather leave this as the responsibility of
+   * caller of this function.
+   *
+   * @image html color_bidirected_graph.png
+   * For example, the image shows 5 nodes numbered from 0 to 4. The connections
+   * shown are bidirectional. After coloring, no two connected nodes have the
+   * same color.
+   *
+   * If deal.II was not installed with package ZOLTAN, this function will
+   * generate an error.
+   *
+   * Zoltan coloring algorithm is run in serial by each processor. Hence all
+   * processors have coloring information of all the nodes.
+   */
+  void color_sparsity_pattern (const SparsityPattern     &sparsity_pattern,
+                               std::vector<int> &color_indices);
```

Comment:
please use `std::vector<unsigned int>` as an output.

Also I think it would make sense to return the total number of colors. 
Of course, one can just loop through `color_indices` to get maximum value: `*std::max_element(color_indices.begin(), color_indices.end());` if Zoltan does not provide this info.